### PR TITLE
Feat/#884 add metadata to Status

### PR DESCRIPTION
**Pull request by BalloonWen**
added metadata to status

### CODE REVIEW

File: status/src/main/java/com/networknt/status/Status.java [hunk 1]
```diff
@@ -258,7 +258,7 @@ public String toString() {
             if (showDescription) {
                 sb.append("\",\"description\":\"" + getDescription());
             }
-            if (showMetadata) {
+            if (showMetadata && getMetadata() != null) {
                 try {
                     sb.append("\",\"metadata\":\"" + Config.getInstance().getMapper().writeValueAsString(getMetadata()));
```
Comment (stevehu) [inline status/src/main/java/com/networknt/status/Status.java:267]: When we convert the metadata map to the string, is the final result still can be parsed as a JSON?

File: handler/src/main/java/com/networknt/handler/LightHttpHandler.java [hunk 2]
```diff
@@ -73,6 +73,24 @@ default void setExchangeStatus(HttpServerExchange exchange, String code, final O
...
 
+    /**
+     * This method is used to construct a standard error status with metadata in JSON format from an error code.
+     *
+     * @param exchange HttpServerExchange
+     * @param code     error code
+     * @param metadata additional metadata info
+     * @param args     arguments for error description
+     */
+    default void setExchangeStatus(HttpServerExchange exchange, String code, Map<String, Object> metadata, final Object... args) {
```
Comment (stevehu) [inline handler/src/main/java/com/networknt/handler/LightHttpHandler.java:84]: Should we call the previous method to reduce the code duplication?

File: status/src/main/java/com/networknt/status/Status.java (hunk 1 above)
Comment (BalloonWen) [reply status/src/main/java/com/networknt/status/Status.java]: now the metadata is appending a JSON instead of a string, and it can be parsed back.

**Comment by stevehu** [comment]
@BalloonWen Could you please update the document for the status with the latest changes? Thanks.

**Comment by BalloonWen** [comment]
> @BalloonWen Could you please update the document for the status with the latest changes? Thanks.

Please take a look and advise https://github.com/networknt/light-doc/pull/234

**Pull request merged**